tests/utils.py

```diff
@@ -149,3 +151,25 @@ def with_dataset(client, name=None, commit_database=False):
     if commit_database:
         client.get_database().register(dataset)
         client.get_database().commit()
+
+
+def retry_failed(fn=None, extended: bool = False):
+    """
+    Decorator to run flaky with same number of max and min repetitions across all tests.
+
+    :param extended: allow more repetitions than usual.
+    """
+
+    def decorate(fn):
+        limit = 20 if extended else 5
+
+        @flaky(max_runs=limit, min_passes=1)
+        @wraps(fn)
+        def wrapper(*args, **kwargs):
+            return fn(*args, **kwargs)
+
+        return wrapper
+
+    if fn:
+        return decorate(fn)
+    return decorate
```

Comment:
I don't think we need 3 level nesting here
```
@flaky(max_runs=5, min_passes=1)
def my_function(...):
    [...]
```

is just syntactic sugar for 
```
my_function = flaky(max_runs=5, min_passes=1)(my_function)
```

so something like
```
def retry_failed(fn=None, extended: bool = False):
    """
    Decorator to run flaky with same number of max and min repetitions across all tests.
    :param extended: allow more repetitions than usual.
    """
    @wraps(func)
    def wrapper(*args, **kwargs):
        func = flaky(max_runs=5, min_passes=1)(func)
        return func(*args, **kwargs)
    return wrapper
```

Should do the trick

Reply:
I think that's doesn't properly handle passing variables to the decorator.
With a solution like that, `@retry_failed` works but `@retry_failed(extended=True)` doesn't.